core/jvm/src/test/scala/zio/logging/LogFilterSpec.scala

```diff
@@ -98,7 +103,20 @@ object LogFilterSpec extends ZIOSpecDefault {
       testFilter(filter, "a.b.c.Exec.exec", LogLevel.Info, Assertion.isFalse) &&
       testFilter(filter, "a.b.c.Exec.exec", LogLevel.Warning, Assertion.isTrue) &&
       testFilter(filter, "e.Exec.exec", LogLevel.Debug, Assertion.isTrue) &&
-      testFilter(filter, "e.f.Exec.exec", LogLevel.Debug, Assertion.isFalse)
+      testFilter(filter, "e.f.Exec.exec", LogLevel.Debug, Assertion.isFalse) &&
+      testFilter(filter, "k.l.Exec.exec", LogLevel.Debug, Assertion.isTrue) &&
+      testFilter(filter, "k.l.m.Exec.exec", LogLevel.Trace, Assertion.isTrue) &&
+      testFilter(filter, "k2.alc.m2.Exec.exec", LogLevel.Trace, Assertion.isTrue) &&
+      testFilter(filter, "k3.alc.m3.Exec.exec", LogLevel.Trace, Assertion.isTrue) &&
+      testFilter(filter, "k3.lc.m3.Exec.exec", LogLevel.Trace, Assertion.isFalse) &&
+      testFilter(filter, "k4.alc.m4.Exec.exec", LogLevel.Trace, Assertion.isTrue) &&
+      testFilter(filter, "k4.al.m4.Exec.exec", LogLevel.Trace, Assertion.isFalse) &&
+      testFilter(filter, "k.l.l.m.Exec.exec", LogLevel.Trace, Assertion.isFalse) &&
+      testFilter(filter, "k.l.l.m.Exec.exec", LogLevel.Debug, Assertion.isTrue) &&
+      testFilter(filter, "q.r.s.t.Exec.exec", LogLevel.Trace, Assertion.isTrue) &&
+      testFilter(filter, "q.r.t.Exec.exec", LogLevel.Trace, Assertion.isTrue) &&
+      testFilter(filter, "q.r.s.u.Exec.exec", LogLevel.Trace, Assertion.isFalse) &&
+      testFilter(filter, "q.r.s.u.Exec.exec", LogLevel.Debug, Assertion.isTrue)
```

Comment:
we should add test where we can see that path without mask take precedence over path with mask

etc
in filter
`"k3.a*.m3" -> LogLevel.Trace`  vs `"k3.alc.m3" -> LogLevel.Warning` 

then "k3.alc.m3.Exec.exec" should have `LogLevel.Warning`

Reply:
rewrote them